crates/node/builder/src/launch/common.rs

```diff
@@ -339,6 +339,10 @@ impl<R> LaunchContextWith<Attached<WithConfigs, R>> {
 
     /// Returns the configured [`PruneConfig`]
     pub fn prune_config(&self) -> Option<PruneConfig> {
+        let config = self.node_config().prune_config();
+        if config.is_some() {
+            return config
+        }
```

Comment:
is this an unrelated fix attempt that ranks cli settings higher than what's in the toml?

I don't think we need to do this because we update the toml settings

https://github.com/paradigmxyz/reth/blob/d395df29fcffbc742cd3e819772cc32271ec2d21/crates/node/builder/src/launch/common.rs#L123-L128

Reply:
not unrelated, blocks syncing op mainnet full node. see docs for `PruningArgs` flags.

Reply:
this will ignore all toml settings which I think is incorrect because this always falls back to the default args if run with --full, no?

Reply:
yeah but that's the usual behaviour for any linux tool: cli args have precedence over files. and we have shipped this help for cli `--full` flag since at least January

https://github.com/paradigmxyz/reth/blob/115a58b327b675336c3c033c1443771e5fc6f10e/crates/node/core/src/args/pruning.rs#L12-L15